src/commands/selection_commands.hpp

```diff
@@ -227,6 +227,26 @@ class dicmdSelectShapesByRegion : public DirectCommandBase
     }
 };
 
+class dicmdSelectAllShapes : public DirectCommandBase
+{
+
+  public:
+    dicmdSelectAllShapes()
+    {
+    }
+
+    virtual void execute()
```

Comment:
Please use the override keyword instead of virtual. If the function does not exist in the parent class you will have a compiler error.